force-app/main/service/ContactService.cls

```diff
@@ -68,54 +68,6 @@ public with sharing class ContactService {
         set;
     }
 
-    /*******************************************************************************************************
-    * @description compares two contacts' addresses
-    * @param con1 a Contact
-    * @param con2 a Contact
-    * @return boolean. true if the Mailing Address fields have changed
-    ********************************************************************************************************/
-    public Boolean isContactAddressChanged(Contact con1, Contact con2) {
-        // if both null, no change
-        if (con1 == null && con2 == null) {
-            return false;
-        }
-
-        // if one null, make sure other has at least one address field set
-        if (con1 == null || con2 == null) {
-            if (con1 == null) {
-                con1 = con2;
-            }
-            return (!isContactAddressEmpty(con1) ||
-                    con1.npe01__Primary_Address_Type__c != null);
-        }
-
-        // both provided, so look for at least one change between address fields
-        Boolean isChanged =
-                !addressServiceInstance.equalsCaseSensitive(con1.MailingStreet, con2.MailingStreet) ||
-                        !addressServiceInstance.equalsCaseSensitive(con1.MailingCity, con2
-                                .MailingCity) ||
-                        !addressServiceInstance.equalsCaseSensitive(con1.MailingState, con2
-                                .MailingState) ||
-                        !addressServiceInstance.equalsCaseSensitive(con1.MailingPostalCode, con2
-                                .MailingPostalCode) ||
-                        !addressServiceInstance.equalsCaseSensitive(con1.MailingCountry, con2
-                                .MailingCountry) ||
-                        !addressServiceInstance.equalsCaseSensitive(con1
```

Comment:
Just to note, the previous logic considered an update to Primary Address Type a "change", even if nothing else was changed or filled in, which is what our new logic also does.

Reply:
Upon further testing, I decided to check for a non-empty address to allow Primary Address Type to cause a change, it was adding unneeded DML just like the insert case that we're trying to resolve!